app/models/manageiq/providers/oracle_cloud/inventory/persister.rb

```diff
@@ -3,28 +3,50 @@ class ManageIQ::Providers::OracleCloud::Inventory::Persister < ManageIQ::Provide
   require_nested :NetworkManager
 
   def initialize_inventory_collections
-    add_collection(cloud, :cloud_tenants) do |builder|
+    add_cloud_collection(:cloud_tenants) do |builder|
       builder.add_default_values(:ems_id => ->(persister) { persister.manager.id} )
     end
-    add_collection(cloud, :flavors)
-    add_collection(cloud, :hardwares)
-    add_collection(cloud, :miq_templates) do |builder|
+    add_cloud_collection(:flavors)
+    add_cloud_collection(:hardwares)
+    add_cloud_collection(:miq_templates) do |builder|
       builder.add_properties(:model_class => ::ManageIQ::Providers::OracleCloud::CloudManager::Template)
     end
-    add_collection(cloud, :operating_systems)
-    add_collection(cloud, :vms)
-    add_collection(cloud, :vm_and_miq_template_ancestry)
-  end
+    add_cloud_collection(:operating_systems)
+    add_cloud_collection(:vms)
+    add_cloud_collection(:vm_and_miq_template_ancestry)
 
-  private
+    add_network_collection(:network_ports)
+    add_network_collection(:cloud_subnets)
+    add_network_collection(:cloud_subnet_network_ports)
+    add_network_collection(:cloud_networks)
+  end
 
   def cloud_manager
     manager.kind_of?(EmsCloud) ? manager : manager.parent_manager
   end
 
+  def network_manager
+    manager.kind_of?(EmsNetwork) ? manager : manager.network_manager
+  end
```

Comment:
These are needed because either the network_manager or the cloud_manager might be the target of the refresh (e.g. `EmsRefresh.refresh(ManageIQ::Providers::OracleCloud::CloudManager.first)` or `EmsRefresh.refresh(ManageIQ::Providers::OracleCloud::NetworkManager.first)`) but we need to always make sure to set the proper ems_id for the various collections.

This is another element that we can move up to the core persister to simplify the providers which do merged refresh.